tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -171,29 +499,26 @@ def __init__(self, target_executors):
         there only is a single participant associated with that placement, as
         would typically be the case with `tff.SERVER`) or lists of target
         executors.
+      intrinsic_strategy_fn: A callable mapping the current executor instance to
+        an instantiation of an IntrinsicStrategy implementation.
 
     Raises:
-      ValueError: If the value is unrecognized (e.g., a nonexistent intrinsic).
+      ValueError: If the target_executors are improper for the given
+        intrinsic_strategy_fn.
     """
-    py_typecheck.check_type(target_executors, dict)
+    py_typecheck.check_callable(intrinsic_strategy_fn)
+    intrinsic_strategy = intrinsic_strategy_fn(self)
+    py_typecheck.check_type(intrinsic_strategy, IntrinsicStrategy)
+    intrinsic_strategy.validate_executor_placements(target_executors)
+    self.intrinsic_strategy = intrinsic_strategy
```

Comment:
WDYT about simply calling this a `strategy`, I think this is going to need to handle more than just federated computations.

Reply:
works for me. are you referring to the ExecutorBase methods?

Reply:
the attribute on the federating_executor.

Reply:
sorry, I was referring to what does "more than federated computations" refer to, but I think we're in sync about this after today's conversation